Fixes - CI, rsync, ansible fact swap

msheiny opened the pull request:
This PR fixes #28, fixes #27, and fixes #26. See individual commit comments for some of the 'Err what the hell!?' questions you might have from looking at the code changes.

I made .. what I probably would call a non-conventional use of molecule in this PR where I'm stringing two scenarios together. On the otherhand, it really addresses the previously shortcomings in full testing the dual gunicorn service model of this role - so I think its a fine trade-off.

Anyways -- to test locally run `make`. Make sure your virtualenv is up to date with requirements -- molecule 2 is still in release-candidate change and has frequent bug fixes right now.

msheiny committed:
Fix rsync capability for local code setup

This is hacky work-around to fix #28 - I continually hit issues using the built-in
synchronize task between two local directories and was unable to resolve. I
attempted to throw the develop version in place to solve this to no avail.
Normally I would file an upstream issue on this but I'm kind of short in time
and this task doesnt need to be idempotent soo I'm going to stick with the
`command` workaround for now.

msheiny committed:
Expand CI testing story

Added a few CI fixes/expansions:
* Adds idempotence test
* Adds a second scenario (ci_rsync) that is designed to be used for two things.
One, it is supposed to be used right after the ci scenario to fix issue #26.
Second, it tests the modules rsync functionality to ensure that it doesnt
regress after issue #28 popped up.

msheiny committed:
Bump molecule and friends!

msheiny committed:
Address ansible fact being changed too early

This fixes #27 by moving the ansible fact swap logic as a handler so it'll only
run if everything else ran successfully.

msheiny committed:
Molecule tweaks - idempotence, linting, doc-strings

msheiny committed:
Remove lookup for CI compat.

The lookup pipe was failing in CI for some reason -- didn't have time to
traced-down since removing it works fine. I originally added the pipe when I was
trying to debug issues with synchronize the control system which didnt help
solve the mystery.

msheiny committed:
Ensure CI related tasks dont affect idempotence

msheiny committed:
Fix for cases when uid != 1000

Because we are performing volume mounting under docker, and the way temp
directories permissions can vary from system to system, the node container can
really only run successfully as a non-privileged user when the current user's id
equals 1000. In any other scenario, lets just run the container with root perms.
This came up in CI particularly.

conorsch commented on molecule/ci/tests/test_basic.py:
```diff
@@ -1,4 +1,4 @@
-def test_mainpage(Command):
+def test_mainpage(host):
```
Was not familiar with the `host` module—handy! :+1:

msheiny replied:
You start to get deprecation warnings if you don't use `host` on recent `testinfra` versions

conorsch commented on handlers/main.yml:
```diff
@@ -16,3 +16,10 @@
+    dest: /etc/ansible/facts.d/active_gunicorn_svc.fact
+    section: django
+    option: name
+    value: "{{ django_stack_active_gcorn_svc }}"
```
Nit: the task name could be more descriptive here. Also, handlers typically have all lowercase names, but I don't see that enforced anywhere.

Consider forcing root:root and mode 0644 on this file, since its contents affect role execution.

conorsch committed:
Hardens local host facts for gunicorn workers

Forcing root:root and mode 0644, since the contents of this file affect
role execution.

conorsch reviewed (approved):
Pretty slick, mate. The more robust CI is great, and the service-swap logic is creative. Methinks the role description on GitHub is no longer accurate, however:

> Minimal ansible role for django development 

:wink:
